packages/sdk/src/types/network/payloads/payloads.ts

```diff
@@ -321,6 +322,11 @@ export type SetBehavior = Payload & {
 	behaviorType: BehaviorType;
 };
 
+export type SetClientAction = Payload & {
+	type: 'set-client-action';
+	clientAction: any;

```

Comment:
Should this be a union type of the client action handlers we are adding so that this is a bit more strongly typed?  That is what we do with the collider geometry types.

Reply:
Yes I was being lazy. It's a little different here because clientAction is a `class` rather than a `type`, so `ClientAction` is the base class rather than a union type. But it should work.

---
In reply to: [316834322](https://github.com/microsoft/mixed-reality-extension-sdk/pull/425#discussion_r316834322) [](ancestors = 316834322)

Reply:
Or, I could make it a type instead. That might be more convenient, and inline with what we do elsewhere.

---
In reply to: [316847203](https://github.com/microsoft/mixed-reality-extension-sdk/pull/425#discussion_r316847203) [](ancestors = 316847203,316834322)

Reply:
Whichever you prefer.  I just like having stuff strongly typed like this.

Reply:
Made it a type union.

---
In reply to: [316855457](https://github.com/microsoft/mixed-reality-extension-sdk/pull/425#discussion_r316855457) [](ancestors = 316855457)

Reply:
Thanks.